Purchases/Networking/RCHTTPClient.m

```diff
@@ -179,16 +186,18 @@ - (NSMutableURLRequest *)createRequestWithMethod:(NSString *)HTTPMethod
         NSError *JSONParseError;
         NSData *body = [NSJSONSerialization dataWithJSONObject:requestBody options:0 error:&JSONParseError];
         if (JSONParseError) {
-            RCLog(@"Error creating request JSON: %@", requestBody);
+            RCErrorLog(@"Error creating request JSON: %@", requestBody);
+            return nil;
         }
         request.HTTPBody = body;
     }
     return request;
 }
 
 - (void)assertIsValidRequestWithMethod:(NSString *)HTTPMethod body:(NSDictionary *)requestBody {
-    NSParameterAssert(!([HTTPMethod isEqualToString:@"GET"] && requestBody));
     NSParameterAssert(([HTTPMethod isEqualToString:@"GET"] || [HTTPMethod isEqualToString:@"POST"]));
+    NSParameterAssert(!([HTTPMethod isEqualToString:@"GET"] && requestBody));
+    NSParameterAssert(!([HTTPMethod isEqualToString:@"POST"] && !requestBody));
```

Comment:
this is a new check, but it looks like we don't have any places where we make POST requests without a body, and I can't think of a reason why you might want to.